src/SFA.DAS.Reservations.Application/Reservations/Commands/CreateReservationCommandHandler.cs

```diff
@@ -1,24 +1,58 @@
-﻿using System.Threading;
+﻿using System;
+using System.Linq;
+using System.Threading;
 using System.Threading.Tasks;
 using MediatR;
+using Newtonsoft.Json;
+using SFA.DAS.Reservations.Application.Reservations.Services;
 using SFA.DAS.Reservations.Application.Validation;
+using SFA.DAS.Reservations.Infrastructure.Api;
+using SFA.DAS.Reservations.Models;
 
 namespace SFA.DAS.Reservations.Application.Reservations.Commands
 {
     public class CreateReservationCommandHandler : IRequestHandler<CreateReservationCommand, CreateReservationResult>
     {
         private readonly IValidator<CreateReservationCommand> _validator;
+        private readonly IApiClient _apiClient;
+        private readonly IHashingService _hashingService;
 
-        public CreateReservationCommandHandler(IValidator<CreateReservationCommand> validator)
+        public CreateReservationCommandHandler(
+            IValidator<CreateReservationCommand> validator, 
+            IApiClient apiClient,
+            IHashingService hashingService)
         {
             _validator = validator;
+            _apiClient = apiClient;
+            _hashingService = hashingService;
         }
 
-        public async Task<CreateReservationResult> Handle(CreateReservationCommand request, CancellationToken cancellationToken)
+        public async Task<CreateReservationResult> Handle(CreateReservationCommand command, CancellationToken cancellationToken)
         {
-            await _validator.ValidateAsync(request);
+            var validationResult = await _validator.ValidateAsync(command);
+            if (!validationResult.IsValid())
+            {
+                throw new ArgumentException(
+                    "The following parameters have failed validation", 
+                    validationResult.ValidationDictionary
+                        .Select(c => c.Key)
+                        .Aggregate((item1, item2) => item1 + ", " + item2));
+            }
 
-            return new CreateReservationResult();
+            var decodedAccountId = _hashingService.DecodeValue(command.AccountId);
+            var apiRequest = new CreateReservationApiRequest
+            {
+                AccountId = decodedAccountId,
+                StartDate = command.StartDate
+            };
+
+            var reservationJson = await _apiClient.CreateReservation(decodedAccountId, JsonConvert.SerializeObject(apiRequest));
```

Comment:
Can the API client be changed to take a generic type where we can set the Reservation?